src/components/__internal__/abstract-custom-field/abstract-custom-field.jsx

```diff
@@ -81,48 +82,57 @@ export default function AbstractCustomField(props) {
       id={ids.input}
       labelId={ids.label}
       label={props.label}
+      name={props.name}
       readOnly={props.readOnly}
       required={props.required}
       tooltip={props.tooltip}
     >
-      <input
-        autoComplete={props.autoComplete}
-        aria-autocomplete={props.ariaProps.autocomplete}
-        aria-controls={props.ariaProps.controls}
-        aria-haspopup={props.ariaProps.haspopup}
-        aria-expanded={props.ariaProps.expanded}
-        aria-invalid={invalidDueToProps ? 'true' : undefined}
-        aria-describedby={`${ids.tooltip} ${ids.validation}`}
-        defaultValue={props.defaultValue}
-        className={inputClassName()}
-        disabled={props.disabled}
-        id={ids.input}
-        role={props.inputRole}
-        max={props.max}
-        maxLength={props.maxLength}
-        min={props.min}
-        name={props.name}
-        onBlur={props.onBlur}
-        onChange={props.onChange}
-        onClick={props.onClick}
-        onFocus={props.onFocus}
-        onKeyDown={props.onKeyDown}
-        onKeyUp={props.onKeyUp}
-        onInput={props.onInput}
-        placeholder={props.placeholder}
-        readOnly={props.readOnly}
-        ref={props.inputRef}
-        required={props.required}
-        step={props.step}
-        style={{ '--numIcon': numIcons() }}
-        type={props.type}
-        value={props.value}
-      />
-      <div className={styles['icon-container']}>
-        { showInvalidIcon() }
-        { showClear() }
-        { showIcon() }
-      </div>
+      <Control
+        labelledBy={ids.label}
+        validationMessage={props.errorText}
+        validationMessageId={ids.validation}
+      >
+        <div style={{ position: 'relative' }}>
```

Comment:
I think this div container is better encapsulated as part of `FormControlIcons` or `ControlIcons`. But I want to do a few more cell controls before doing that particular refactor.